front/app/components/FilterBoxes/TopicsFilter.tsx

```diff
@@ -91,35 +123,91 @@ const TopicsFilter = memo<Props & InjectedLocalized>(
-            {topics
-              .filter((topic) => !isError(topic))
-              .map((topic: ITopicData) => (
-                <Topic
-                  key={topic.id}
-                  data-id={topic.id}
-                  onMouseDown={removeFocusAfterMouseClick}
-                  onClick={handleOnClick}
-                  className={`e2e-topic ${
-                    includes(selectedTopicIds, topic.id) ? 'selected' : ''
-                  }`}
-                >
-                  <T value={topic.attributes.title_multiloc} />
-                </Topic>
-              ))}
+          <Box>
+            <Box className="e2e-topics-filters" aria-live="polite">
+              {topicsWithIdeas
+                .filter((topic) => !isError(topic))
+                .map((topic: ITopicData, index) => {
+                  const postCount = get(
+                    filterCounts,
+                    `topic_id.${topic.id}`,
+                    0
+                  );
+
+                  const topicSelected = selectedTopicIds?.includes(topic.id);
+
+                  if (!showFullList && index > 4) {
+                    // Only show the first 5 topics by default
+                    return null;
+                  }
+
+                  return (
+                    <Topic
+                      key={topic.id}
+                      data-id={topic.id}
+                      onMouseDown={removeFocusAfterMouseClick}
+                      onClick={handleOnClick}
+                      className={`e2e-topic ${topicSelected ? 'selected' : ''}`}
+                      style={{ display: 'flex' }}
+                    >
+                      <Box maxWidth="90%">
+                        <Text
+                          fontSize="s"
+                          m="0px"
+                          color={topicSelected ? 'white' : 'textPrimary'}
+                          wordBreak="break-word"
+                        >
+                          <T value={topic.attributes.title_multiloc} />
+                          <ScreenReaderOnly>
+                            {`${postCount} ${formatMessage(messages.inputs)}`}
+                          </ScreenReaderOnly>
+                        </Text>
+                      </Box>
+                      <Box aria-hidden>{postCount}</Box>
+                    </Topic>
+                  );
+                })}
+            </Box>
+            {topicsWithIdeas.length > 5 && (
+              <Button
+                onClick={() => {
+                  setShowFullList((curentValue) => !curentValue);
+                }}
+                buttonStyle="text"
+                p="0px"
+                mt="12px"
+                fontSize="s"
+              >
+                {showFullList
+                  ? formatMessage(messages.showLess)
+                  : `${formatMessage(messages.showTagsWithNumber, {
+                      numberTags: topicsWithIdeas.length,
+                    })}`}
```

Comment:
```suggestion
{formatMessage(
                  showFullList
                    ? messages.showLess
                    : messages.showTagsWithNumber,
                  {
                    numberTags: topicsWithIdeas.length,
                  }
                )}
```

Slightly cleaner? :)